File: app/src/main/java/com/nytimes/android/external/register/di/AppSchedulerProvider.java
package com.nytimes.android.external.register.di;

import io.reactivex.Scheduler;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

final class AppSchedulerProvider implements SchedulerProvider {
    @Override 
    public Scheduler getMainThread() {
        return AndroidSchedulers.mainThread();
    }

    @Override 
    public Scheduler getComputationThread() {
        return Schedulers.computation();
    }

    @Override 
    public Scheduler getIoThread() {
        return Schedulers.io();
    }
}

File: app/src/main/java/com/nytimes/android/external/register/di/AppSchedulerProvider.kt
package com.nytimes.android.external.register.di

import io.reactivex.Scheduler
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers

internal class AppSchedulerProvider : SchedulerProvider {

    override fun getMainThread(): Scheduler {
        return AndroidSchedulers.mainThread()
    }

    override fun getComputationThread(): Scheduler {
        return Schedulers.computation()
    }

    override fun getIoThread(): Scheduler {
        return Schedulers.io()
    }
}
